File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):
    cnt = 0
    l = len(p)
    array = []
    
    for i in range(len(t)-l+1):
        array.append(int(t[i:i+l]))
    for i in array:
        if i <= int(p):
            cnt+=1
    return cnt

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.swift
import Foundation

extension String {
    subscript (_ range: Range<Int>) -> String {
        let fromIndex = self.index(self.startIndex, offsetBy: range.startIndex)
        let toIndex = self.index(self.startIndex, offsetBy: range.endIndex)
        return String(self[fromIndex..<toIndex])
    }
}

func solution(_ t:String, _ p:String) -> Int {
    var result = 0
    
    let cntP = p.count
    let cntT = t.count
    
    let long = cntP < cntT ? cntT : cntP
    let short = cntP > cntT ? cntT : cntP

    for i in 0...long-short {
        if let num = Int(t[i..<i+cntP]), let numP = Int(p) {
            if num <= numP {
                result += 1
            }
        }
    }
    return result
}